### PR TITLE
Update go-ectd to v0.4.6

### PR DESCRIPTION
**Pull request by satnam6502**
In response to issue #5779
Updating the master to use etcd v.2.0.5 is in PR  #5780

### CODE REVIEW

**Comment by satnam6502** [comment]
Some tests don't pass yet. Will investigate.
```
--- FAIL: TestSyncPodsDeletesWithNoPodInfraContainer (0.00s)
	kubelet_test.go:114: expected []string{"list", "list", "list", "list", "inspect_container", "inspect_container", "list", "inspect_container", "inspect_container", "stop", "create", "start", "inspect_container", "create", "start", "list", "inspect_container", "inspect_container"}, got []string{"list", "list", "list", "stop", "create", "start", "inspect_container", "create", "start", "list", "inspect_container", "inspect_container", "list", "inspect_container", "inspect_container", "list", "inspect_container", "inspect_container"}
```

**Comment by dchen1107** [comment]
Can you run a e2e test?

**Comment by thockin** [comment]
Is there no way to just update the go-etcd client lib without updating go, protobuf, etcd, and friends.

**Comment by satnam6502** [comment]
Well in the original PR I just had the go-etcd changes.
But I noticed that this is compile  our unit/integration tests due to a missing dependency.
So I added that dependency and then re-ran godeps.

**Comment by satnam6502** [comment]
Most of the e2e tests pass, with a few of the usual suspects failing.
```
Summarizing 4 Failures:

[Fail] Shell [It] tests that guestbook.sh passes 
/usr/local/google/home/satnam/gocode/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/shell.go:67

[Fail] Monitoring [It] verify monitoring pods and all cluster nodes are available on influxdb using heapster. 
/usr/local/google/home/satnam/gocode/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/monitoring.go:235

[Fail] Shell [It] tests that services.sh passes 
/usr/local/google/home/satnam/gocode/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/shell.go:67

[Fail] Shell [It] tests that .guestbook.sh.swp passes 
/usr/local/google/home/satnam/gocode/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/shell.go:67

Ran 29 of 31 Specs in 3361.691 seconds
FAIL! -- 25 Passed | 4 Failed | 2 Pending | 0 Skipped F0323 11:10:42.032463   15762 driver.go:87] At least one test failed
```

**Comment by satnam6502** [comment]
@dchen1107 I can't see how to reduce the dependencies. Are you willing to go with this as-is or is there something else we can try? @a-robinson in case he can help from his previous experience.

**Comment by a-robinson** [comment]
If those are all legitimate dependencies of the new package, I'm not sure we can do much. You could try manually eliminating the code that's causing them to be pulled in (it's just one line in each of the first couple layers of dependencies, IIRC from looking at it this afternoon).

Also, why'd you drop back to 0.4.6 instead of 2.0.5?

**Comment by satnam6502** [comment]
I had the wrong version to start off with because I confused it with the
most recent version of etcd which is 2.0.5 but the most recent version of
go-etcd is 0.4.6

Cheers,

Satnam

**Comment by dchen1107** [comment]
Why you upgrade GoVersion from 1.4.1 to 1.4.2 in this PR?

**Comment by satnam6502** [comment]
Oh. I did not notice that I had. Ugh. I guess that is because I have go 1.4.2 on my desktop machine where I did the build?

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -63,14 +58,29 @@
 			"Rev": "b965b613227fddccbfffe13eae360ed3fa822f8d"
 		},
 		{
+			"ImportPath": "github.com/codegangsta/inject",
```
Comment (dchen1107) [inline Godeps/Godeps.json]: Why introduce codegangsta/inject package here? I couldn't find go-etcd has depency on this one.

File: Godeps/Godeps.json [hunk 2]
```diff
@@ -185,6 +195,11 @@
 			"Rev": "588cb435e59ee8b6c2795482887755841ad67207"
 		},
 		{
+			"ImportPath": "github.com/go-martini/martini",
```
Comment (dchen1107) [inline Godeps/Godeps.json]: why include go-martini/martini too?

**Comment by satnam6502** [comment]
I wondered about that too. Let me try to do this again afresh by deleting my Godpes directory and re-generating afresh. However, it is strange that when I updated just go-etcd I got a small number of Godeps changes that I expected... but then when I also did the go get for the CoreOs httpttypes that it suddenly seemed to pull in lots of other dependencies.

**Comment by satnam6502** [comment]
I am closing this PR and making a new one based on a fresh repo.

**Pull request closed without merging**